cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/RunTestnet.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Testnet.Test.RunTestnet
+  ( hprop_run_testnet
+  ) where
+
+import           Prelude
+
+import           Cardano.Api (BlockNo (..), ChainTip (..))
+import           Cardano.CLI.Type.Output (QueryTipLocalStateOutput (..))
+import qualified Data.Aeson as A
+import qualified Data.ByteString.Lazy.Char8 as B
+import           Data.Default.Class (def)
+import           Data.Maybe (isJust)
+import           System.Exit (ExitCode (..))
+import qualified System.Process as IO
+
+import           Cardano.Testnet
+import           Testnet.Property.Util (integrationWorkspace)
+import           Testnet.Start.Types (GenesisOptions (..))
+
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras as H
+import           Testnet.Process.Run (execCli',mkExecConfig)
+
+-- | Execute me with:
+-- @DISABLE_RETRIES=1 cabal test cardano-testnet-test --test-options '-p "/Testnet produces blocks/"'@
+hprop_run_testnet :: H.Property
+hprop_run_testnet = integrationWorkspace "run-testnet" $ \tmpDir -> H.runWithDefaultWatchdog_ $ do
+
+  let shelleyOptions = def { genesisEpochLength = 200 }
+      testnetOptions = def
+
+  conf <- mkConf tmpDir
+  TestnetRuntime {testnetNodes} <- cardanoTestnetDefault testnetOptions shelleyOptions conf
+
+  -- There should only be one producer node among three
+  TestnetNode
+    { nodeProcessHandle
+    , nodeSprocket
+    } <- case testnetNodes of
+      [producerNode, _relayNode1, _relayNode2] -> do
+        -- Only SPO nodes have pool keys
+        H.diff (isJust . poolKeys <$> testnetNodes) (==) [True, False, False]
+        pure producerNode
+      _ -> H.failure
+  -- Let the node run for 30 seconds, to let problems time to happen
+  H.threadDelay 30_000_000 -- microseconds
+
+  -- Check that blocks have been produced on the chain
+  execConfig <- mkExecConfig tmpDir nodeSprocket 42
```

Comment:
```suggestion
  execConfig <- mkExecConfig tmpDir nodeSprocket testnetMagic
```
It'll be 42 most of the cases, but you can get `testnetMagic` from `TestnetOptions`.